optimism/src/preimage_oracle.rs

```diff
@@ -116,19 +115,19 @@ impl PreImageOracle {
         // We need to map 3, 4, 5, 6 in the child process
         cmd.fd_mappings(vec![
             FdMapping {
-                parent_fd: hint_server.0.writer.as_raw_fd(),
+                parent_fd: unsafe { OwnedFd::from_raw_fd(hint_server.0.writer.as_raw_fd()) },
```

Comment:
I guess this is because some `optimism` library was upgraded? But why is it an unsafe call now/is it sound?

Reply:
yes, a dependency upgradaded its public api to require OwnedFd, modeling as a rype that the code already took ownership of the fd. the unsafety represents that the fd is going to be closed on drop, making the original raw fd dangling